src/modules/flow/http-client/http-client.c

```diff
@@ -422,13 +476,20 @@ common_get_process(struct sol_flow_node *node, void *data, uint16_t port,
     }
 
     sol_http_params_init(&params);
-    if (mdata->accept && !sol_http_param_add(&params,
-        SOL_HTTP_REQUEST_PARAM_HEADER("Accept", mdata->accept))) {
-        SOL_WRN("Failed to set query params");
-        r = -ENOMEM;
-        goto err;
+
+    if (mdata->accept) {
+        if (!sol_http_param_add(&params,
+            SOL_HTTP_REQUEST_PARAM_HEADER("Accept", mdata->accept))) {
+            SOL_WRN("Failed to set query params");
+            r = -ENOMEM;
+            goto err;
+        }
+
+        if (strstr(mdata->accept, "text/stream"))
```

Comment:
hum... this is tricky, we should check based on order. The accept "proper handling" would also account for the "q=" parameter, basically you split it to an array of struct { struct sol_str_slice mime; double q; } and then order by `q`.

@ceolin did you ever implement that? maybe it's a function to add to `sol-http.h`, it would take a slice and returns a `struct sol_vector` (not pointer!) with the sorted vector. Our function would check if text/stream is the highest (first) priority.